components/Menu.php

```diff
@@ -223,9 +223,12 @@ public function createMenuItemObject($menuItem)
         $object->specialIsActive = ($menuItem->special AND $menuItem->special->active());
         $object->specialDaysRemaining = optional($menuItem->special)->daysRemaining();
 
-        $object->menuPrice = $object->specialIsActive
-            ? $menuItem->special->getMenuPrice($menuItem->menu_price)
-            : $menuItem->menu_price;
+        $object->menuPrice = $menuItem->menu_price;
+
+        if($object->specialIsActive){
+            $object->oldPrice = $menuItem->menu_price;
```

Comment:
Rename `oldPrice` to `menuPriceBeforeSpecial` and put spaces around () in the if statement. Also create a temporary variable for menuPriceBeforeSpecial 

ex. `$object->menuPriceBeforeSpecial = $object->menuPrice = $menuItem->menu_price;`